src/set_theory/game/nim.lean

```diff
@@ -0,0 +1,257 @@
+/-
+Copyright (c) 2020 Fox Thomson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fox Thomson
+-/
+import set_theory.game.impartial
+import set_theory.ordinal
+import data.set
+import logic.basic
+
+universes u v
+
+/-!
+# Nim and the Sprague-Grundy theorem
+
+This file contains the defition for nim for any ordinal `O`. In the game of `nim O₁` both players
+may move to `nim O₂` for any `O₂ < O₁`.
+We also define a Grundy number for an impartial game `G` and prove the Sprague-Grundy theorem, that
+`G` is equivalent to `nim (Grundy_value G)`.
+-/
+
+open pgame
+
+local infix ` ≈ ` := equiv
+
+/-- The defineition of nim, nim O is the game where each player can move to nim O' where O' < O
+    If O is finite then this game can be veiwed as a pile of stones where each player can take a
+  positie number of stones from it on their turn -/
```

Comment:
I think some explanation is needed, either here or in the module-doc, that this definition is talking about single-heap Nim. Any introduction you'll read elsewhere to Nim will describe the multiple-heap version, of course.